### PR TITLE
fix the CMAKE order for issue #630

### PR DESCRIPTION
**Pull request by StevenPuttemans**
Ensure that the double pass through CMakeLists.txt files do not generate double output
More details discussed in issue https://github.com/Itseez/opencv_contrib/issues/630

### CODE REVIEW

**Comment by StevenPuttemans** [comment]
@alalek these are the 4 modules in opencv_contrib suffering from this issue

**Comment by StevenPuttemans** [comment]
@alalek it seems that the proposed fixed where not as straightforward as considered :P it leads to a shitload of errors. Any idea why?

File: modules/cvv/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+set(the_description "Debug visualization framework")
+ocv_define_module(cvv opencv_core opencv_imgproc opencv_features2d ${CVV_LIBRARIES} WRAP python)
```
Comment (alalek) [inline modules/cvv/CMakeLists.txt]: This module requires QT5, so there 2 lines should be placed after next `if` block

File: modules/cnn_3dobj/CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,6 @@
+set(the_description "CNN for 3D object recognition and pose estimation including a completed Sphere View on 3D objects")
+ocv_define_module(cnn_3dobj opencv_core opencv_imgproc opencv_viz opencv_highgui OPTIONAL WRAP python)
```
Comment (alalek) [inline modules/cnn_3dobj/CMakeLists.txt]: This module requires caffe. See the line 27: `if(HAVE_CAFFE)`

File: modules/hdf/CMakeLists.txt [hunk 3]
```diff
@@ -19,6 +19,9 @@ else()
   find_package(HDF5)
 endif()
 
+set(the_description "Hierarchical Data Format I/O")
+ocv_define_module(hdf opencv_core WRAP python)
```
Comment (alalek) [inline modules/hdf/CMakeLists.txt]: Should be after block with `ocv_module_disable(hdf)` statement (line 26/29)

**Comment by alalek** [comment]
:+1:

File: modules/cvv/CMakeLists.txt (hunk 1 above)
Comment (mshabunin) [inline modules/cvv/CMakeLists.txt]: I think this file should be reverted:

- variable `CVV_LIBRARIES` is used before being set
- other preparation steps (`add_definitions`, `include_directories`, set compiler flags)  do not affect some subtargets, probably leading to #659

**Pull request merged**